File: backend/pmgl-api/src/main/java/fr/mecanique/api/pmgl/pmgl_api/admin/dto/AdminDTO.java
package fr.mecanique.api.pmgl.pmgl_api.admin.dto;

import fr.mecanique.api.pmgl.pmgl_api.account.bean.Account;
import fr.mecanique.api.pmgl.pmgl_api.admin.enums.AdminRole;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Date;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AdminDTO {

    private Long idAdmin;

    private String zone;

    private Long accountId;

    private String refAdmins;

    private AdminRole role;

    private Date creationDate;
}

File: front/pmgl-app/src/app/component/admin-package/beans/admin-dto.ts
import { AccountDTO } from "../../customer-package/beans/account-dto.model";

export type AdminRole = 'SUPER_ADMIN' | 'ADMIN' | 'COMMERCIAL' | 'TECH';

export interface AdminDTO {
  idAdmin: number;
  zone?: string | null;
  account: AccountDTO;
  refAdmins?: string | null;
  role: AdminRole;
  creationDate: string; // ISO
}
